tiledb/sm/rest/curl.cc

```diff
@@ -571,6 +571,11 @@ Status Curl::make_curl_request_common(
       *curl_code = tmp_curl_code;
     }
 
+    /* If there is a write error we should not attempt a retry */
```

Comment:
1) IMO we should combine with the if block above for clarity
2) IIUC the only condition leading to a retry now will be an HTTP status in one of the `retry_http_codes_`. Is that correct? If so I think we should add a comment to clarify.